pkg/queue/concurrency_state_test.go

```diff
@@ -140,20 +143,50 @@ func TestConcurrencyStateHandlerParallelOverlapping(t *testing.T) {
 	}
 }
 
-func TestConcurrencyStateRequestHeader(t *testing.T) {
+func TestConcurrencyStateTokenRefresh(t *testing.T) {
+
+	oldTokenPath := filepath.Join(t.TempDir(), "secret")
+	if err := os.WriteFile(oldTokenPath, []byte("0123456789"), 0700); err != nil {
+		t.Fatal(err)
+	}
+
+	c := NewConcurrencyEndpoint("freeze-proxy", oldTokenPath)
+	oldToken := fmt.Sprint(c.token.Load())
```

Comment:
this feels a bit too low-level to me, we shouldn't care - in the test - about the internal private fields of the struct. Better to test that the correct header is received in the test server.

Reply:
I was leaning the other way (that what we wanted to test was that the token was refreshed, and adding a test server was adding too much to the test), but not something I feel too strongly about. Will update.